lib/qobj.c

```diff
@@ -25,14 +25,15 @@
 #include "hash.h"
 #include "log.h"
 #include "qobj.h"
+#include "jhash.h"
 
 static pthread_rwlock_t nodes_lock;
 static struct hash *nodes = NULL;
 
 static unsigned int qobj_key(void *data)
 {
 	struct qobj_node *node = data;
-	return (unsigned int)node->nid;
+	return jhash(&node->nid, sizeof(node->nid), 0);
```

Comment:
This is somewhat pointless since qobj `nid`s are already random

Reply:
except of course nid is a uint64_t and we are downcasting it to a unsigned int.  As such there is no guarantee that we would not end up with the same value for the key.  I don't think this is unreasonable to do.